OpenSim/Examples/Plugins/CoupledBushingForceExample/CoupledBushingForce.cpp

```diff
@@ -50,37 +50,23 @@ CoupledBushingForce::~CoupledBushingForce()
 /**
  * Default constructor.
  */
-CoupledBushingForce::CoupledBushingForce() : Force(),
-    _body1Name(_body1NameProp.getValueStr()),
-    _body2Name(_body2NameProp.getValueStr()),
-    _locationInBody1(_locationInBody1Prop.getValueDblVec()),
-    _orientationInBody1(_orientationInBody1Prop.getValueDblVec()),
-    _locationInBody2(_locationInBody2Prop.getValueDblVec()),
-    _orientationInBody2(_orientationInBody2Prop.getValueDblVec())
+CoupledBushingForce::CoupledBushingForce() : Force()
 {
-    constructMatricesFromProperties();
     setNull();
 }
 
 /* Convenience constructor */
-CoupledBushingForce::CoupledBushingForce( std::string body1Name, SimTK::Vec3 point1, SimTK::Vec3 orientation1,
-                  std::string body2Name, SimTK::Vec3 point2, SimTK::Vec3 orientation2,
-                  SimTK::Mat66 stiffnessMat, SimTK::Mat66 dampingMat):  Force(),
-    _body1Name(_body1NameProp.getValueStr()),
-    _body2Name(_body2NameProp.getValueStr()),
-    _locationInBody1(_locationInBody1Prop.getValueDblVec()),
-    _orientationInBody1(_orientationInBody1Prop.getValueDblVec()),
-    _locationInBody2(_locationInBody2Prop.getValueDblVec()),
-    _orientationInBody2(_orientationInBody2Prop.getValueDblVec())
+CoupledBushingForce::CoupledBushingForce( const PhysicalFrame& frame1,
+                                          const PhysicalFrame& frame2,
+                                        SimTK::Mat66 stiffnessMat,
+                                        SimTK::Mat66 dampingMat) : Force()
 {
     setNull();
-    constructMatricesFromProperties();
-    _body1Name = body1Name;
-    _body2Name = body2Name;
-    _locationInBody1 = point1;
-    _orientationInBody1 = orientation1;
-    _locationInBody2 = point2;
-    _orientationInBody2 = orientation2;
+    constructInfrastructure();
+
+    updConnector<PhysicalFrame>("frame1").set_connected_to_name(frame1.getName());
+    updConnector<PhysicalFrame>("frame2").set_connected_to_name(frame2.getName());
```

Comment:
If the `CoupledBushingForce` only depends on the *names* of the `PhysicalFrame` arguments, should those arguments just be the names rather than the binary object? That would allow some out-of-order model building as can be done through XML, I think.